bitcoin/src/blockdata/block.rs

```diff
@@ -249,20 +249,14 @@ impl Block {
 
         let cb = self.coinbase().ok_or(Bip34Error::NotPresent)?;
         let input = cb.input.first().ok_or(Bip34Error::NotPresent)?;
-        let push = input.script_sig.instructions_minimal().next().ok_or(Bip34Error::NotPresent)?;
-        match push.map_err(|_| Bip34Error::NotPresent)? {
-            script::Instruction::PushBytes(b) => {
-                // Check that the number is encoded in the minimal way.
-                let h = b
-                    .read_scriptint()
-                    .map_err(|_e| Bip34Error::UnexpectedPush(b.as_bytes().to_vec()))?;
-                if h < 0 {
-                    Err(Bip34Error::NegativeHeight)
-                } else {
-                    Ok(h as u64)
-                }
-            }
-            _ => Err(Bip34Error::NotPresent),
+        let push = input.script_sig.instructions_minimal()
+            .next()
+            .ok_or(Bip34Error::NotPresent)?
+            .map_err(to_bip34_error)?;
+        match (push.script_num(), push.push_bytes().map(|b| b.read_scriptint())) {
+            (Some(num), Some(Ok(_)) | None) => Ok(num.try_into().map_err(|_| Bip34Error::NegativeHeight)?),
```

Comment:
Caveat: I'm far from an expert on script.

Correct me if I'm wrong but it seems like there is something unusual here, `script_num` and `push_bytes` returning `Some` is mutually exclusive but the pattern includes `(Some, Some)`.

Also it seems odd that we use `instructions_minimal` and then later use `to_bip34_error` that checks for minimal push?

Reply:
Mad, thanks for the answer. I'll review again.